test/python/test_transpiler.py

```diff
@@ -101,3 +104,94 @@ def test_pass_cx_cancellation(self):
         resources_after = dag_circuit.count_ops()
 
         self.assertNotIn('cx', resources_after)
+
+    def test_lookahead_mapper_doesnt_modify_mapped_circuit(self):
+        """Test that lookahead mapper is idempotent.
+
+        It should not modify a circuit which is already compatible with the
+        coupling map, and can applied repeatedly without modifying the circuit.
+        """
+
+        qr = QuantumRegister(3)
+        circuit = QuantumCircuit(qr)
+        circuit.cx(qr[0], qr[2])
+        circuit.cx(qr[0], qr[1])
+        original_dag = DAGCircuit.fromQuantumCircuit(circuit)
+
+        # Create coupling map which contains all two-qubit gates in the circuit.
+        coupling_map = Coupling(couplinglist=[[0, 1], [0, 2]], regname=qr.name)
```

Comment:
why do we need a regname

Reply:
:+1: Removed. This pass had been keeping the creg structure of the input dag, rather than mapping out to a single device creg `q`. Fixed in 492dbb81695e0a407bc487a3786a8ade286e841e.